### PR TITLE
OCPBUGS-18590: Enable kdump for IBM Power and Z libvirt jobs

### PR DESCRIPTION
**Pull request by jaypoulz**
This patch does two main things, and is organized by commits:

The first is that it updates the logic for kdump used by the realtime gcp jobs to move it towards a more generic, shared directory. Additionally, a bunch of the logic is updated to address a few bugs I found:
1. The gather step didn't seem to work as intended. it was missing a trailing '/' for the `$ oc cp` command to work properly
2. The gather step didn't correctly count the number of nodes it was pulling content from.
3. The gather step depended on the the name of the debug pod to be consistent between calls to oc debug. At least on the multi-arch CI clusters, this is not the case, since a random hash is appended each time you create one.

Additionally, here are some enhancements I made:
1. The configure step now accepts input to the /etc/sysconfig/kdump file so it can be configured for the specific job.
2. I've added logs to the configuration steps so you can see exactly what got created.

Lastly, I've enabled kdump by default for all the multi-arch libvirt jobs. This resulted in one important caveat:
- The gather logic now needs access to the multi-arch DNS container so that it can correctly resolve clusters. Ideally, I could wrap the step so that this gets injected into the pod without it being in the ref itself. However, when I tested this, it didn't seem to work, so this may be a limitation in CI.

In order to test this, I'm going to purposely crash nodes on the rehearsal jobs to show the output is gather as expected.

### CODE REVIEW

**Comment by jaypoulz** [comment]
/pj-rehearse run periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-gcp-ovn-rt-upgrade

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-rt-upgrade-test

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-rt-upgrade-test

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-rt-upgrade-test eriodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-rt-upgrade-test periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
This should be ready for final review. In the `ci/rehearse/periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le` job, I intentionally crashed the 2 workers and kill the running tests just to show that the kdump output would be detected and collected for both nodes.

In the periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-rt-upgrade-test job, I crashed 1 worker and let the rest of the job run to completion.

In the periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x job, I crashed no workers and hoped to see the output of the test be relatively normal.

**Comment by jaypoulz** [comment]
/hold

Setting a hold until I have a final ACK from the RT/LVM folks.

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.13-ocp-e2e-ovn-remote-libvirt-ppc64le

File: ci-operator/step-registry/ipi/libvirt/pre/ipi-libvirt-pre-chain.yaml [hunk 1]
```diff
@@ -2,10 +2,23 @@ chain:
   as: ipi-libvirt-pre
   steps:
   - ref: ipi-conf-libvirt
+  - ref: ipi-conf-debug-kdump-configure-logs
   - chain: ipi-install-libvirt
   env:
   - name: ARCH
   - name: BRANCH
+  - name: CRASH_KERNEL_MEMORY
```
Comment (Prashanth684) [inline ci-operator/step-registry/ipi/libvirt/pre/ipi-libvirt-pre-chain.yaml:10]: is this only impacting multi-arch libvirt or x86 as well?

Comment (jaypoulz) [reply ci-operator/step-registry/ipi/libvirt/pre/ipi-libvirt-pre-chain.yaml:10]: Only us - this is the full list of affected tests:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pj-rehearse/openshift/release/42882/a46435fc0206afc1787b84161b370251afe3b567

The ipi-libvirt-pre chain is only used by us.

Comment (jaypoulz) [reply ci-operator/step-registry/ipi/libvirt/pre/ipi-libvirt-pre-chain.yaml:10]: This is the workflow used by the CRC guys:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/code-ready/crc/e2e/code-ready-crc-e2e-workflow.yaml

**Comment by jaypoulz** [comment]
periodic-ci-openshift-multiarch-master-nightly-4.13-ocp-e2e-ovn-remote-libvirt-ppc64le is hitting the same etcd flake we know about, but is proof that the Power jobs aren't affected.
periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x is showing issues with installs, but this seems pervasive to the jobs across the platform right now. Raised an issue in slack with the team.

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse  periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by eggfoobar** [comment]
/lgtm

Thanks @jaypoulz!

**Comment by jaypoulz** [comment]
/pj-rehearse ack

I've checked the s390x rehearsal and it's running as expected now. I don't see any signs of memory pressure. Going to check this in to get the logs for any crashes this weekend.

**Pull request merged**